examples/multisig_transfer.ts

```diff
@@ -0,0 +1,150 @@
+// FIXME: remove this check once the Zones .bind(env) fix is merged
+const hostname = Deno.env.get("TEST_CTX_HOSTNAME");
+const portRaw = Deno.env.get("TEST_CTX_PORT");
+if (!hostname || !portRaw) {
+  throw new Error("Must be running inside a test ctx");
+}
+
+import { createKeyMulti } from "../deps/polkadot/util-crypto.ts";
+import * as C from "../mod.ts";
+import * as T from "../test_util/mod.ts";
+import * as U from "../util/mod.ts";
+
+type KeyringPair = typeof T.alice;
+
+// Signatories must be sorted
+const signatories: Uint8Array[] = [
+  T.alice.publicKey,
+  T.bob.publicKey,
+  T.charlie.publicKey,
+].sort();
+const THRESHOLD = 2;
+const multisigPublicKey = createKeyMulti(signatories, THRESHOLD);
+
+const env = C.Z.env();
+
+// Transfer initial balance (existential deposit) to multisig address
+U.throwIfError(await addBalanceToMultisigAddress().bind(env)());
+
+// Create proposal
+U.throwIfError(await createOrApproveMultisigProposal({ sender: T.alice }).bind(env)());
+
+// Get the key of the timepoint
+const key = C.keyPageRead(T.polkadot)("Multisig", "Multisigs", 1, [multisigPublicKey])
+  .access(0)
+  .access(1);
+
+// Get the timepoint itself
+const maybe_timepoint = await C.entryRead(T.polkadot)("Multisig", "Multisigs", [
+  multisigPublicKey,
+  key,
+])
+  .access("value")
+  .access("when")
+  .bind(env)();
+
+U.throwIfError(
+  await createOrApproveMultisigProposal({ sender: T.bob, maybe_timepoint }).bind(env)(),
+);
+
+// check T.dave new balance
+console.log(
+  U.throwIfError(
+    await C.entryRead(T.polkadot)("System", "Account", [T.dave.publicKey]).bind(env)(),
+  ),
+);
+
+function addBalanceToMultisigAddress() {
+  return C.extrinsic({
+    client: T.polkadot,
+    sender: {
+      type: "Id",
+      value: T.alice.publicKey,
+    },
+    palletName: "Balances",
+    methodName: "transfer",
+    args: {
+      value: 2_000_000_000_000n,
+      dest: {
+        type: "Id",
+        value: multisigPublicKey,
+      },
+    },
+  })
+    .signed((message) => ({
+      type: "Sr25519",
+      value: T.alice.sign(message),
+    }))
+    .watch(function(status) {
+      console.log(status);
+      if (C.TransactionStatus.isTerminal(status)) {
+        this.stop();
+      }
+    });
+}
+
+interface CreateOrApproveProposalProps {
```

Comment:
Given that this interface isn't reused, should we inline this constraints of `createOrApproveMultisigProposal`? Ie.

```diff
function createOrApproveMultisigProposal(
- { sender, maybe_timepoint }: CreateOrApproveProposalProps,
+ publicKey: Uint8Array,
+ sign: C.M.SignExtrinsic,
+ maybeTimepoint?: bigint,
) {
```